src/monitor/monitor_netlink.c

```diff
@@ -26,6 +26,9 @@
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #define __USE_GNU /* needed for struct ucred */
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE /* needed by musl to enable struct ucred */
+#endif
```

Comment:
On July 14, 2016 at 1:40:26 pm +02:00, lslebodn <<notifications@github.com>> wrote:

> In src/monitor/monitor_netlink.c <https://github.com/SSSD/sssd/pull/4#discussion_r70790779>:
> > @@ -26,6 +26,9 @@
> > #include <sys/types.h>
> > #include <sys/ioctl.h>
> > #define __USE_GNU /* needed for struct ucred */
> > +#ifndef _GNU_SOURCE
> > +#define _GNU_SOURCE /* needed by musl to enable struct ucred */
> > +#endif
> 
> 
> We should not define _GNU_SOURCE ourselves.
> I would remove define __USE_GNU as well an include "config.h" in the beginning of file.
> _GNU_SOURCE is automatically part of config.h if available on system.
> 
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/SSSD/sssd/pull/4/files/bd4c5186e85619c0d6fdc36690b311734e552fe4#r70790779>, or mute the thread <https://github.com/notifications/unsubscribe/AAipUq_v-JpJywlwHntkZhsJqQ-Jd-4sks5qViAqgaJpZM4JMLmi>.
> 
> 
> 
> 
> Ah, will change that. I didn't really find any good way to check for the existence of struct ucred, but if config.h supplies that info, then its better to just include that instead.

Mvh
Daniel Kuehn, 073-181 577 2
Data@UrService
Hemsida: <http://www.dataaturservice.se>